kin-recovery/recovery/src/main/java/kin/recovery/BackupAndRestoreManager.java

```diff
@@ -19,22 +19,23 @@
 	public static final String PUBLIC_ADDRESS_EXTRA = "publicAddressExtra";
 
 	private final CallbackManager callbackManager;
-	private final KinClient kinClient;
+	private KinClient kinClient;
 	private Activity activity;
 
-	public BackupManager(@NonNull final Activity activity, @NonNull KinClient kinClient) {
+	public BackupAndRestoreManager(@NonNull final Activity activity) {
```

Comment:
why `KinClient` moved from ctor?

Reply:
Because kinClient can be changed then it is not appropriate for backup manager to hold it.
After this change they should send it as a parameter to backup or restore methods and in that way it will be always up to date

Reply:
ok, so why we save it as a local field? should be used one time

Reply:
For the onSuccess of registerRestoreCallback.
We need to extract the KinAccount from the KinClient.

Reply:
> LGTM overall, I have a comment regarding sample structure though,
> while using architecture is a good practice, I think it adds unnecessary complexity to a sample app,
> I think it will be easier to follow without the added layers of MVP.
> Moreover, I think it will be even better to separate the account creation from the backup code (and UI) meaning have some UI (i.e. activity) for creating the necessary account on the blockchain, and then a dedicated activity for the backup.

Talked about it with Segev and we are thinking that in this case there is a justification for a presenter or at least some kind of a place which will handle all of the logic code.
Regarding the account creation - Yossi Segev designed the UI and I also talked with him about your comment and he says that currently we should keep it that way.
Maybe later will change a bit the sample app.

Reply:
missed that... :eyes: